core/src/test/java/org/infinispan/configuration/UpdatableConfigurationTest.java

```diff
@@ -0,0 +1,29 @@
+package org.infinispan.configuration;
+
+import static org.testng.AssertJUnit.assertEquals;
+
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.testng.annotations.Test;
+
+@Test(groups = "functional", testName= "configuration.UpdatableConfigurationTest")
+public class UpdatableConfigurationTest {
+
+   public void testUpdateConfiguration() {
+      ConfigurationBuilder builder = new ConfigurationBuilder();
+      builder.expiration().maxIdle(1000).lifespan(2000);
+      Configuration configuration = builder.build();
+
+      // Check that attributes are correct before the update
+      assertEquals(1000, configuration.expiration().maxIdle());
+      assertEquals(2000, configuration.expiration().lifespan());
+
+      builder = new ConfigurationBuilder();
+      builder.expiration().maxIdle(3000); // Lifespan will be set to the default
```

Comment:
Why does `maxIdle(3000)` change lifespan?
Or are you saying that even attributes that have the default values are included in the update?

And I'd include a test with an incompatible attribute as well.